backend/dissemination/workbooklib/federal_awards.py

```diff
@@ -213,10 +213,13 @@ def generate_federal_awards(dbkey, year, outfile):
     addls = _fix_addl_award_identification(Cfda, cfdas, dbkey)
     set_range(wb, "additional_award_identification", addls)
 
-    (prefixes, extensions) = _fix_pfixes(cfdas)
+    (prefixes, extensions, full_cfdas) = _fix_pfixes(cfdas)
     set_range(wb, "federal_agency_prefix", prefixes)
     set_range(wb, "three_digit_extension", extensions)
 
+    # We need a `cfda_key` as a magic column for the summation logic to work/be checked.
+    set_range(wb, "cfda_key", full_cfdas, conversion_fun=str)
```

Comment:
In federal awards workbooks we use this formula `=IF(OR(B{0}="",C{0}=""),"",CONCATENATE(B{0},".",C{0}))`  to build `cfda_key` but according to the logic [here](https://github.com/GSA-TTS/FAC/pull/2563/files#diff-ff2dabbd25da7c9de0277ab7292bd1acd3afc8d8541e68a377bf9e736e659628R127) an item in `full_cfdas` is not garanteed to always be the concatenation of item in `prefixes` and item in `extensions` due to `else "000"`. This could lead to random bugs